### DISCUSSION TITLE
How to use the L293d motor shield ?

### DISCUSSION THREAD

**Discussion by royhair** [giuliopons/vintagephone#2] [General]
Hi. Great ideas in this project. I’m trying to get the bells ringing on my vintage phone but I’m news and struggling with how to operate the motor shield without an arduino. 
I’ve connected the 12v output from the step up board to the external power pins and I’ve removed the jumper on the motor shield. My Wemos is putting out something as I’ve tested using a multimeter. But now do l connect the output of the wemos to the motor shield? And why two pins?  I know this is basic but it's new to me. 

I've tried connecting various pins to the Wemos but the m1 to m4 outputs only a hundredth of the 12v input 
Could a kind person please supply a diagram and instructions on how to connect things up?
Thanks!
[PXL_20220811_062331206.jpg](https://user-images.githubusercontent.com/74099594/184075451-c44ef209-d84b-4b59-a745-2a05cbbd16a1.jpg)

Output from Wemos: 
![20220811_101200.gif](https://user-images.githubusercontent.com/74099594/184101496-119f5bb6-390e-4cdc-a3c0-08a1c347a8a2.gif)

**Comment by giuliopons** [comment]
Hi, your shield is different from mine.

![image](https://user-images.githubusercontent.com/1871627/184904967-257c20ec-52ed-40c0-8ee2-37756651a0fa.png)

We need to have an alternate voltage applied to the bells. 
This kind of shields are commonly used to drive motors with an external power source, so with two pins from Wemos we can turn on and off alternatively the power of the output pins A+ A- of the shield.

For example, we put LOW to IN1 and HIGH to IN2 we got the output A- to 0 and A+ to 1. 
Then we can invert the values (LOW to IN2 and HIGH to IN1) to get A+ to 0 and A- to 1.
By doing this we have an alternate voltage between A- and A+ and we can use it to feed to the coil with alternatively voltage from -12 V to +12 V.

I've used pins D2 and D3 from Wemos to drive the motor shield.

Hope this helps!

**Reply by royhair** [reply]
Thanks for your reply, Giulio. My Wemos is connected like yours and I am using pins 2 and 3.

That's the instructions I'm following but I don't know how to get the motor shield to put out the 12v AC. It doesn't seem to be happening however I connect it. Online instructions for the shield are geared toward driving motors, not quick switching to produce AC current.
